src/app/app.component.ts

```diff
@@ -18,12 +19,29 @@ export class AppComponent {
     constructor(
         private readonly _router: Router,
         public readonly stateService: StateService,
-        public viewportService: ViewportService
+        public viewportService: ViewportService,
+        private readonly _ref: ChangeDetectorRef,
+        private readonly _breakpointObserver: BreakpointObserver
     ) {}
 
+    ngOnInit(): void {
+        this._breakpointObserver
+            .observe([Breakpoints.Small, Breakpoints.Handset])
+            .subscribe((state: BreakpointState) => {
+                if (state.matches) {
+                    this.stateService.setDrawerOpen(false);
+                    this._ref.detectChanges();
+                } else {
+                    this.stateService.setDrawerOpen(true);
+                    this._ref.detectChanges();
+                }
+            });
+    }
+
```

Comment:
I'm wondering why we need this extra logic to detect when we move into mobile view. I would expect the temporary drawer to be closed when the drawerOpen variable is false without having to set it to false again. If I take this breakpoint logic out and go from big screen to little screen i see this:
![image](https://user-images.githubusercontent.com/29152776/89564717-6af5c700-d7eb-11ea-980a-72cc836e5244.png)
![image](https://user-images.githubusercontent.com/29152776/89564745-75b05c00-d7eb-11ea-81b3-19c39a0d2bb4.png)

The temporary drawer is open despite `open=false`. This seems like a bug in the component where we need to handle some sort of onchange when the variant changes.

Reply:
I've found the issue and Joe you're right, this bug is stemming from within our component library.

The DrawerLayout component is updating the internal state of the Drawer when the variant is set to persistent, and this causes the Drawer's `open` state to get out of sync with the App's `open` input variable.

I'll add a fix to the component library and see if we can get that merged in before this PR.

Reply:
Awesome!

Reply:
New PR just opened here:

https://github.com/pxblue/angular-component-library/pull/132

This PR refactors the internal state management logic of the Drawer component so that it won't override what a user passes in as the `open` input prop.

Reply:
When this PR goes in, the AppComponent will still need to manually specify that the Drawer `open=false` whenever the variant transitions from one type to the other.